src/OpenTelemetry.Api/Context/BaggageContext.cs

```diff
@@ -0,0 +1,241 @@
+﻿// <copyright file="BaggageContext.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenTelemetry.Context
+{
+    /// <summary>
+    /// Baggage context.
+    /// </summary>
+    public readonly struct BaggageContext : IEquatable<BaggageContext>
+    {
+        private static readonly RuntimeContextSlot<BaggageContext> RuntimeContextSlot = RuntimeContext.RegisterSlot<BaggageContext>("otel.baggage_context");
+        private static readonly Dictionary<string, string> EmptyBaggage = new Dictionary<string, string>();
```

Comment:
IReadOnlyDictionary?

Reply:
If I switch it here, I can't cheat and pass either EmptyBaggage or this.baggage into the Dictionary ctor when I want to make a copy. It should only ever be returning to the user an IReadOnlyDictionary tho. Good enough for now? Later on, I want to remove the Dictionary completely and use something more efficient for the copy behavior the spec is asking for.

Reply:
Definitely, 🐹 is not a blocker at all 😄